Frameworks/CoreGraphics/DWriteWrapper.mm

```diff
@@ -479,4 +479,170 @@ HRESULT _DWriteFontGetBoundingBoxesForGlyphs(
     });
 
     return ret;
-}
\ No newline at end of file
+}
+
+#pragma region CGFontCreateWithDataProvider helpers
+
+/**
+ * Custom implementation of IDWriteFontFileStream that implements Read in terms of an underlying CFDataRef
+ */
+class DWriteFontBinaryDataStream : public RuntimeClass<RuntimeClassFlags<WinRtClassicComMix>, IDWriteFontFileStream> {
+protected:
+    InspectableClass(L"Windows.Bridge.DirectWrite", TrustLevel::BaseTrust);
+
+public:
+    DWriteFontBinaryDataStream() {
+    }
+
+    HRESULT RuntimeClassInitialize(CGDataProviderRef dataProvider) {
+        _data.reset(CGDataProviderCopyData(dataProvider));
+
+        // Just use current time for _lastWriteTime
+        FILETIME fileTime;
+        SYSTEMTIME systemTime;
+
+        GetSystemTime(&systemTime);
+        // Returns 0 on failure, which is probably good enough for WinObjC purposes?
+        SystemTimeToFileTime(&systemTime, &fileTime);
```

Comment:
not sure if GetSystemTimeToFileTime has the same behavior. I realize that 0 will change to 0 in the below code anyway, but I find special-casing it explicitly to be a comparatively minor optimization that, for me, muddies up the code path.

---
In reply to: [88171264](https://github.com/Microsoft/WinObjC/pull/1386#discussion_r88171264) [](ancestors = 88171264)